### PR TITLE
Map Count with HotRestart enabled and atleast one index

### PR DESCRIPTION
**Pull request by chanmol1999**

### CODE REVIEW

**Comment by chanmol1999** [comment]
@erosb Please review.

File: hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/PhoneHomeTest.java [hunk 1]
```diff
@@ -167,5 +167,41 @@ public void testMapCountWithAtleastOneQueryCache() {
...
+        parameters = phoneHome.phoneHome(true);
+        assertEquals(parameters.get("mpaoict"), "0");
+
+        Map<String, String> map1 = node.hazelcastInstance.getMap("hazelcast");
+        parameters = phoneHome.phoneHome(true);
+        assertEquals(parameters.get("mpaoict"), "0");
+
+        map1.put("1", "hazelcast");
+        parameters = phoneHome.phoneHome(true);
+        assertEquals(parameters.get("mpaoict"), "1");
```
Comment (erosb) [inline hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/PhoneHomeTest.java:183]: I can't see where is the index added.

File: hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/PhoneHomeIntegrationTest.java [hunk 2]
```diff
@@ -47,9 +47,12 @@ public void testMapMetrics() {
         PhoneHome phoneHome = new PhoneHome(node, "http://localhost:8080/ping");
         Map<String, String> map1 = hz.getMap("hazelcast");
         Map<String, String> map2 = hz.getMap("phonehome");
+        map2.put("1", "hazelcast");
```
Comment (erosb) [inline hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/PhoneHomeIntegrationTest.java:50]: This isn't an index. It is a map entry.

File: hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java [hunk 3]
```diff
@@ -77,4 +79,18 @@ private long countMapWithAtleastOneQueryCache(Node node) {
         }).count();
 
     }
+
+    private long countMapWithAtleastOneIndex(Node node) {
+        return maps.stream().filter(distributedObject -> !(node.hazelcastInstance.getMap(distributedObject.getName()).isEmpty())).count();
```
Comment (erosb) [inline hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java:84]: This counts the non-empty maps, not the maps with indexes. Please refer to the reference manual about indexing: https://docs.hazelcast.org/docs/latest-dev/manual/html-single/#indexing-queries

**Pull request merged**